### PR TITLE
fix: vite build issue when using self directive in manifest.json

### PR DESCRIPTION
**Pull request by jpozo20**
Community guidelines:

- [x] I have read and understand the
      [contributing guidelines](https://github.com/extend-chrome/.github/blob/master/.github/CONTRIBUTING.md)
      and
      [code of conduct](https://github.com/extend-chrome/.github/blob/master/.github/CODE_OF_CONDUCT.md)

This PR contains:

- [x] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

List any relevant issue numbers: #279 

### Description

When the manifest.json contains the 'self' directive either in content_security_policy.extension_pages or content_security_policy.sandbox, then `vite build` fails with `Error: unable to parse manifest code`. 

After debugging for a while, I found out this is because the manifest is transformed into a file with a named default export I guess because of the single quotes needed in the CSP. This was causing the node tree to not contain a `Literal` as the `decodeManifest` function expects, instead the value of the json was found inside a `TemplateElement`.  

The PR adds an `AcornTemplateElement` that is verified if the value from the Literal comes empty and throws the exception if the template element is also empty. It was my first time fixing something in a node a library so it took me a while to setup the environment to actually debug. I will add tests covering the error and the solution later.

### CODE REVIEW

**Comment by jacksteamdev** [comment]
@jpozo20 Thanks for digging in and contributing 🤗 

LGTM! A very simple snapshot test in `tests/mv3` should be enough. If it passes locally, don't worry about CI passing. Some of the serve tests are flaky.

**Comment by jpozo20** [comment]
@jacksteamdev I copied the basic-ts folder and added two tests. I don't really get how snapshots work outside of components, but the second test will fail with the new code since it expects to throw an error if the 'self' directive is found, while both will work if you comment the assignment of the `templateElement` in the `simple` visitor since RPCE will throw the error.

**Comment by jacksteamdev** [comment]
@jpozo20 Yeah, it's a kind of alternative use of snapshot testing. We need to add a section on tests in the upcoming docs. Basically, the snapshot testing strategy validates the output files and fails when the output of RPCE changes. 

The functions `build` and `serve` run the [Vite JavaScript API](https://vitejs.dev/guide/api-javascript.html) and perform some setup and cleanup tasks. They will fail if Vite throws an error.

The function `testOutput` checks that all the files in the manifest are present in the output files, and uses snapshots to verify that the list of output files has not changed and that the contents of the output files did not change.

I treat the snapshot files as code. If a change shows up in Git that I don't expect, I know that my changes are causing some unintended side effects. Previously I was using a more traditional testing strategy with explicit expectations, but that failed in two ways: unexpected changes would slip through my tests, and it took a loooong time to update tests to handle changes. With snapshots, it's easy to know when something changes, and easy to update tests when those changes are correct.

File: tests/mv3/vite-self-directive-in-csp/build.test.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  it("should work if the 'self' directive is not found", async () => {
+    const result = await build(__dirname)
+    await testOutput(result)
+  })
+
+  it("should fail if the 'self' directive is found", async () => {
+    const runBuild = async () => await build(__dirname, 'vite.csp.ts')
+    await expect(runBuild).rejects.toThrowError('unable to parse manifest code')
+  })
+})
```
Comment (jacksteamdev) [inline tests/mv3/vite-self-directive-in-csp/build.test.ts:17]: We just need one test that passes if the 'self' directive is present. The original `basic-ts` test handles the case where there is no CSP.

**Review by jacksteamdev — CHANGES_REQUESTED** [review]
Overall this looks good! Hopefully, I was able to explain the testing strategy successfully.

**Comment by jacksteamdev** [comment]
I'm getting ready to make some big changes to the structure of this repo, so I'm going to merge this PR. Thanks for chasing down this bug and making the fix!

**Pull request merged**